extensions/resource-deployment/src/services/notebookService.ts

```diff
@@ -0,0 +1,68 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import { NotebookInfo } from '../interfaces';
+import { isString } from 'util';
+import * as os from 'os';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as nls from 'vscode-nls';
+const localize = nls.loadMessageBundle();
+
+export interface INotebookService {
+	launchNotebook(notebook: string | NotebookInfo): void;
+}
+
+export class NotebookService implements INotebookService {
+	/**
+	 * Copy the notebook to the user's home directory and launch the notebook from there.
+	 * @param notebook the path of the notebook
+	 */
+	launchNotebook(notebook: string | NotebookInfo): void {
+		const notebookRelativePath = this.getNotebook(notebook, () => process.platform);
+		const notebookFullPath = path.join('../', notebookRelativePath);
+		if (notebookRelativePath && fs.existsSync(notebookFullPath)) {
+			const targetFileName = this.getTargetNotebookName(notebookFullPath, os.homedir(), fs.existsSync);
+			fs.copyFileSync(notebookFullPath, targetFileName);
+			vscode.commands.executeCommand('vscode.open', vscode.Uri.file(targetFileName));
+		} else {
+			vscode.window.showErrorMessage(localize('resourceDeployment.notebookNotFound', 'The notebook {0} does not exist', notebookFullPath));
+		}
+	}
+
+	getNotebook(notebook: string | NotebookInfo, getPlatform: () => string): string {
```

Comment:
I assume you're passing in getPlatform as a parameter to help with testing? It's an odd thing to do otherwise and seems like we should find a better way to mock stuff like this

Reply:
i made a new service so that I can mock all the operations like copyFile, fileExists as well as platform. also added more test cases.